### ISSUE TITLE
[BUG] Duplicate methods/fields generated

### ISSUE THREAD

**Issue by TheGeekPharaoh** [OpenAPITools/openapi-generator#12187]
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [X] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

Duplicate fields and methods are being generated when generating JAX-RS spec server

When generating Java server code using the jaxrs-spec generator, model classes are generated with duplicate fields, methods, constructors, etc.  I'm not positive, but I believe it may be due to the fact that the OpenAPI spec I'm generating off of contains $ref declarations.

##### openapi-generator version

5.4.0

##### OpenAPI declaration file content or url

The OpenAPI spec in question is here:  

https://github.com/TheGeekPharaoh/asc4j/blob/main/src/main/resources/enhanced_app_store_connect_api_1.8_openapi.json

##### Generation Details

openapi-generator generate -i openapi.json \
    --api-package net.odyssi.asc4j.jaxrs \
    -g jaxrs-spec \
    --model-package net.odyssi.asc4j.model \
    --additional-properties=booleanGetterPrefix=is,dateLibrary=java8,openApiNullable=false,serializableModel=true,hideGenerationTimestamp=true,generateBuilders=true,sourceFolder=src/main/java,interfaceOnly=true,useSwaggerAnnotations=false

##### Steps to reproduce

Running the above command generates the malformed Java output

**Comment by borsch** [comment]
@TheGeekPharaoh I found duplicated enum constants in `SubscriptionStatusUrlVersion`. This is because of how enum vars from spec are converted to java enum vars

https://github.com/OpenAPITools/openapi-generator/blob/f92f8f1e58c59a429295509d8c5df32b100cde22/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java#L1527

so `v1` and `V1` values in spec will result into `V1` value in java enum.

If you really want to have enum with such values I can propose you to look at importing of custom models. https://openapi-generator.tech/docs/usage/#type-mappings-and-import-mappings. Using this you can manually create desired model and import it into code generated from specification

**Comment by TheGeekPharaoh** [comment]
Yes, I saw that, too.  That’s the least of my concerns, however.  The generated model classes have hundreds of duplicated fields and constructors.  That’s the bigger issue.  In one file I saw the same constructor generated 4 times.


    On Thursday, April 21, 2022, 04:24:47 AM EDT, Oleh Kurpiak ***@***.***> wrote:  
 
 


@TheGeekPharaoh I found duplicated enum constants in SubscriptionStatusUrlVersion. This is because of how enum vars from spec are converted to java enum vars

https://github.com/OpenAPITools/openapi-generator/blob/f92f8f1e58c59a429295509d8c5df32b100cde22/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java#L1527

so v1 and V1 values in spec will result into V1 value in java enum.

If you really want to have enum with such values I can propose you to look at importing of custom models. https://openapi-generator.tech/docs/usage/#type-mappings-and-import-mappings. Using this you can manually create desired model and import it into code generated from specification

**Comment by borsch** [comment]
@TheGeekPharaoh  can you please give example of java class where you observe such thing? I'll check

**Comment by TheGeekPharaoh** [comment]
The attached file (AgeRatingDeclaration.java) was generating using this command:

openapi-generator generate -i openapi.json \
--api-package net.odyssi.asc4j.jaxrs \
-g jaxrs-spec \
--model-package net.odyssi.asc4j.model \
--additional-properties=booleanGetterPrefix=is,dateLibrary=java8,openApiNullable=false,serializableModel=true,hideGenerationTimestamp=true,generateBuilders=true,sourceFolder=src/main/java,interfaceOnly=true,useSwaggerAnnotations=false


It has multiple copies of the same constructor.  It appears to be an issue with the jaxrs-spec generator.  When I generate client code using the java/resteasy generator it works fine.  This command works:

openapi-generator generate -i openapi.json \
--api-package net.odyssi.asc4j.jaxrs \
-g java \
--model-package net.odyssi.asc4j.model \
--additional-properties=booleanGetterPrefix=is,dateLibrary=java8,openApiNullable=false,hideGenerationTimestamp=true,generateBuilders=true,sourceFolder=src/main/java,interfaceOnly=true,useSwaggerAnnotations=false,library=resteasy




[AgeRatingDeclaration.java.txt](https://github.com/OpenAPITools/openapi-generator/files/8534499/AgeRatingDeclaration.java.txt)

**Comment by borsch** [comment]
@TheGeekPharaoh I've fixed this issue - https://github.com/OpenAPITools/openapi-generator/pull/12202
Currently I did only code generation fix for 5.4 patch version

FYI - similar issue was already reported - https://github.com/OpenAPITools/openapi-generator/issues/11884 and fix was provided only to master branch(which will be a base for 6.0.0 version). That fix contains improvements(i.e. generate lombok's like SuperBuilder). So consider migration to never version if you want to get those improvements

**Comment by borsch** [comment]
I did a cherrypick of https://github.com/OpenAPITools/openapi-generator/pull/11892 from master to 5.4 branch
